### PR TITLE
Enable style checks

### PR DESCRIPTION
**Pull request by braingram**
Style checks (via pre-commit) have not been checking any files in this repo as part of the CI.
See this recent run on main:
https://github.com/spacetelescope/stcal/actions/runs/11017955113/job/30597119856#step:10:63
With all checks showing:
```
check for added large files..........................(no files to check)Skipped
```
If I run them locally I see a large number of automatic changes:
36 files changed, 1091 insertions(+), 942 deletions(-)
and even after these see 79 ruff errors, many from codespell and 6 from reporeview.

In an effort to check something this PR:
- fixes the `check-style` job to check files
- disables all failing ruff and pre-commit rules (maybe we can re-enable these one-by-one in follow-up PRs)

With this PR the checks run:
https://github.com/spacetelescope/stcal/actions/runs/11130795981/job/30930929199?pr=293#step:10:49

## Tasks
- [ ] update or add relevant tests
- [ ] update relevant docstrings and / or `docs/` page
- [ ] Does this PR change any API used downstream? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [ ] run regression tests with this branch installed (`"git+https://github.com/<fork>/stcal@<branch>"`)
    - [ ] [`jwst` regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/jwst.yml) 
    - [ ] [`romancal` regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml)

<details><summary>news fragment change types...</summary>

- ``changes/<PR#>.apichange.rst``: change to public API
- ``changes/<PR#>.bugfix.rst``: fixes an issue
- ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
</details>

### CODE REVIEW

File: tox.ini
Comment (braingram) [inline tox.ini:33]: Since the CI runs this with a trailing `--` tox was skipping the important `--all-files` and instead trying to run on only non-committed files of which there were none.

**Review by zacharyburnett — APPROVED** [review]

File: tox.ini
Comment (braingram) [reply tox.ini:33]: We could leave it there to always be green 😆

File: .pre-commit-config.yaml
Comment (kmacdonald-stsci) [inline .pre-commit-config.yaml:15]: Why are these removed?  Don't we want these checks?  And why comment them out, instead of deleting them?

File: .pre-commit-config.yaml
Comment (kmacdonald-stsci) [inline .pre-commit-config.yaml:60]: Why is this commented out and not simply removed?

File: .pre-commit-config.yaml
Comment (kmacdonald-stsci) [inline .pre-commit-config.yaml:25]: Why is this commented out instead of just removed?

File: pyproject.toml
Comment (kmacdonald-stsci) [inline pyproject.toml:124]: Why are these commented out instead of just removed?

**Review by kmacdonald-stsci — APPROVED** [review]

File: .pre-commit-config.yaml
Comment (braingram) [reply .pre-commit-config.yaml:15]: Doh, I had clicked auto-merge so this got merged as soon as you approved it. I can follow this up with a PR to remove them if that's what you'd like.

I commented them out because these are all failing and likely have been for some time. The PR that enabled them had a bug in the workflow where it ran the checks but never supplied any files to check (so it was checking nothing). See https://github.com/spacetelescope/stcal/pull/293#discussion_r1783271648 for a comment on where this bug is fixed. Here's a PR that uncomments all the checks to show the extent of the failures:
https://github.com/spacetelescope/stcal/pull/294
The reported diff for the checks is over 5000 lines long:
https://github.com/spacetelescope/stcal/actions/runs/11142692931/job/30966194873?pr=294
(and doesn't fix all the failures and many require manual updates).

I chose to not remove them because I assumed that the maintainers want these checks to run. If that's not the case I'd say we remove them entirely. If the re-enabling the checks are preferred I was thinking it would be easiest to enable them perhaps one-by-one since each will require file changes.

Comment (kmacdonald-stsci) [reply .pre-commit-config.yaml:15]: I'm not sure what's the best approach for this.  It could be good to have them commented out, so we have explicit documentation for what is not used, but removing them to clean up the script could be good, too.  I just wanted to make sure you wanted the comments left in the scripts.

**Pull request merged**
